primitives/sr-primitives/src/lib.rs

```diff
@@ -451,6 +436,27 @@ impl From<&'static str> for DispatchError {
 	}
 }
 
+/// This type specifies the outcome of dispatching a call to a module.
+///
+/// In case of failure an error specific to the module is returned.
+pub type DispatchOutcome = Result<(), DispatchError>;
+
+/// The outcome of inclusion of an extrinsic into a block.
+///
+/// This type is typically used in the context of `BlockBuilder` to signal that the extrinsic
+/// in question cannot be included. It is fair to say that a valid block doesn't contain any
+/// extrinsic that would have had a negative inclusion outcome. On successful inclusion this type
+/// supplies the result of the extrinsic dispatch.
+///
+/// Examples of reasons preventing inclusion in a block:
+/// - More block weight is required to process the extrinsic than is left in the block being built.
```

Comment:
I wouldn't name weight here and just say block resources. Weight is Palette specific while I suppose these error types are.

Reply:
Ahhh, that's a good point. I am mistakingly assumed that we are in the pallete context.